### PR TITLE
Update googletest to 1.15.2

### LINKED ISSUE #35540: Newer release available `com_google_googletest`: v1.15.2 (current: a4ab0ab)
Package Name: com_google_googletest@a4ab0abb93
Current Version: a4ab0ab@2020-09-10
Available Version: v1.15.2@2024-07-31
Upstream releases: https://github.com/google/googletest/releases

### PR DESCRIPTION
**Pull request by asedeno**
Fix #35540 

This picks up nearly four years of changes in googletest.

googletest has updated to abseil flags[^1].
> The most notable difference is that unrecognized flags result in a flag parsing error,
> and are not returned to the user though a modified argc/argv, unless they appear
> after the positional argument delimiter ("--").
> 
> For example, to pass a non-Abseil flag, you would have to do
> ./mytest --gtest_color=false -- --myflag=myvalue

After this upgrade, to set Envoy's CLI arguments in tests requires passing them after `--`.
For example, 
```
bazel test //test/common/common:logger_test --test_arg="--" --test_arg="-l trace"
```

Per googletest documentation[^2], it is strongly recommended that  death tests
have names ending in `DeathTest` rather than simply `Test`.

Some death tests behave better when run in a threadsafe manner[^3].
This is a trade-off that costs speed. The flag has been set in tests where it made a difference.

When building with bazel and abseil, googletest uses re2[^4]. The switch to re2 means
that `.` does not match newline by default. This prefixes regex that need it with `(?s)`
to change that behavior[^5].

Risk Level: low — test only
Testing: Updates

[^1]: https://github.com/google/googletest/commit/25dcdc7e8bfac8967f20fb2c0a628f5cf442188d
[^2]: https://github.com/google/googletest/blob/main/docs/advanced.md#death-test-naming
[^3]: https://github.com/google/googletest/blob/main/docs/advanced.md#death-test-styles
[^4]: https://github.com/google/googletest/blob/main/docs/advanced.md#death-test-styles
[^5]: https://github.com/google/re2/wiki/syntax#flags

### CODE REVIEW

**Comment by asedeno** [comment]
cc: @phlax @adisuissa @yanavlasov
x-ref #33219

File: test/fuzz/fuzz_runner.cc [hunk 1]
```diff
@@ -87,7 +87,7 @@ void runCleanupHooks() {
 extern "C" int LLVMFuzzerInitialize(int* argc, char*** argv) {
   // Before parsing gmock flags, set the default value of flag --gmock_verbose to "error".
   // This suppresses logs from NiceMock objects, which can be noisy and provide little value.
-  testing::GMOCK_FLAG(verbose) = "error";
+  GMOCK_FLAG_GET(verbose) = "error";
```
Comment (adisuissa) [inline test/fuzz/fuzz_runner.cc:90]: Should probably be: `GMOCK_FLAG_SET(verbose, "error");`

File: test/integration/protocol_integration_test.cc [hunk 2]
```diff
@@ -4769,7 +4769,8 @@ class AllowForceFail : public Api::OsSysCallsImpl {
   bool fail_ = false;
 };
 
-TEST_P(ProtocolIntegrationTest, HandleUpstreamSocketCreationFail) {
+TEST_P(ProtocolIntegrationDeathTest, HandleUpstreamSocketCreationFail) {
+  GTEST_FLAG_SET(death_test_style, "threadsafe");
```
Comment (adisuissa) [inline test/integration/protocol_integration_test.cc:4773]: Why is this needed here (and in some of the other tests)? I thought it is set once in the test_runner.

**Review by adisuissa — COMMENTED** [review]
Thanks for pursuing this!
I cannot recall which CI test was problematic, but I hope we can get this solved in this PR.

File: test/integration/protocol_integration_test.cc (hunk 2 above)
Comment (asedeno) [reply test/integration/protocol_integration_test.cc:4773]: I did not go digging into why. All I can tell you is that anywhere I set it explicitly, the tests failed without it.

File: test/fuzz/fuzz_runner.cc (hunk 1 above)
Comment (asedeno) [reply test/fuzz/fuzz_runner.cc]: Done.

**Comment by asedeno** [comment]
Fun. Coverage tests are failing because the disk is full. From `df-post`:
```
Filesystem      Size  Used Avail Use% Mounted on                                                                                     
/dev/root        73G   73G   16M 100% /                                                                                                            
tmpfs           7.9G  172K  7.9G   1% /dev/shm                                                                                                     
tmpfs           3.2G  1.1M  3.2G   1% /run                                                                                                         
tmpfs           5.0M     0  5.0M   0% /run/lock                                                                                                    
/dev/sdb15      105M  6.1M   99M   6% /boot/efi                                                                                                    
/dev/sda1        74G  4.1G   66G   6% /mnt                                                                                                         
tmpfs           1.6G   12K  1.6G   1% /run/user/1001                                                                                               
```

**Comment by asedeno** [comment]
clang and gcc are being killed building some targets. We can iterate and keep bumping up the rbe_pools for targets one by one, but I don't like it.

**Comment by phlax** [comment]
> Fun. Coverage tests are failing because the disk is full

hmm - i added the diskspace-hack (ie delete some of the default installed cruft before running) so i would have hoped this didnt happen - lmc again ...

> clang and gcc are being killed building some targets. We can iterate and keep bumping up the rbe_pools for targets one by one, but I don't like it.

i have some tooling to identify which jobs need more mem by profiling a complete run with everything running on max mem workers, but tbh, with most of the problematic tasks already identified, doing what you are doing is probably more reliable and less time consuming

**Comment by phlax** [comment]
so diskspace hack is running and frees up a fair amount of space but evidently not enough

```console
Disk space before cruft removal
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        73G   58G   16G  80% /
tmpfs           7.9G  172K  7.9G   1% /dev/shm
tmpfs           3.2G  1.1M  3.2G   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sda15      105M  6.1M   99M   6% /boot/efi
/dev/sdb1        74G  4.1G   66G   6% /mnt
tmpfs           1.6G   12K  1.6G   1% /run/user/1001
Removing: /opt/hostedtoolcache ...
Removing: /usr/local/lib/android ...
Removing: /usr/local/.ghcup ...
Disk after cruft removal
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        73G   36G   37G  50% /
tmpfs           7.9G  172K  7.9G   1% /dev/shm
tmpfs           3.2G  1.1M  3.2G   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sda15      105M  6.1M   99M   6% /boot/efi
/dev/sdb1        74G  4.1G   66G   6% /mnt
tmpfs           1.6G   12K  1.6G   1% /run/user/1001
```

i guess we just need to `rm -rf` even more paths to free space - i can raise a PR for that now

**Comment by asedeno** [comment]
In the last Mobile/ASAN run, `//test/common:internal_engine_test` failed to compile and it's already set to run on `4core`.

File: mobile/test/common/BUILD [hunk 3]
```diff
@@ -18,6 +18,7 @@ envoy_cc_test(
 envoy_cc_test(
     name = "internal_engine_test",
     srcs = ["internal_engine_test.cc"],
+    rbe_pool = "2core",
```
Comment (phlax) [inline mobile/test/common/BUILD]: mobile CI doesnt use these pools

for ref here are the mobile pools:

https://envoy.cluster.engflow.com/restatus

and the envoy ones:

https://mordenite.cluster.engflow.com/restatus

Comment (phlax) [reply mobile/test/common/BUILD]: also mobile is not set up to use `rbe_pool` so it will either break stuff or at best be ignored

it is, however, OOMing

```console
ERROR: /source/mobile/test/common/BUILD:18:14: Compiling test/common/internal_engine_test.cc failed: (Killed): clang-14 failed: error executing command (from target //test/common:internal_engine_test) 
```

which means there is a bigger issue in terms of mobile workers and resources - probably you will need to raise this on the related channel

**Comment by asedeno** [comment]
Still OOMing. Waiting on further resources.

File: mobile/test/common/extensions/filters/http/platform_bridge/BUILD [hunk 4]
```diff
@@ -31,6 +31,7 @@ envoy_extension_cc_test(
         "platform_bridge_filter_integration_test.cc",
     ],
     extension_names = ["envoy.filters.http.platform_bridge"],
+    rbe_pool = "6gig",
```
Comment (phlax) [inline mobile/test/common/extensions/filters/http/platform_bridge/BUILD:34]: this wont work out of the box - what it actually needs to be is:

```starlark
exec_properties = {"Pool": "6gig"}
```

probably it makes sense to set up a macro - so that you can set it as `rbe_pool` - see eg

https://github.com/envoyproxy/envoy/blob/4c0975228ab82e8c4965f351a4cbf771a027617f/bazel/envoy_library.bzl#L117-L120
